src/chart/generator/marker.cpp

```diff
@@ -64,14 +64,18 @@ Marker::Marker(const Options &options,
 	}
 
 	auto horizontal = options.isHorizontal();
+	auto lineOrCircle = options.geometry == ShapeType::line
+	                 || options.geometry == ShapeType::circle;
+
 	position.x = size.x = getValueForChannel(channels,
 	    ChannelId::x,
 	    data,
 	    stats,
 	    options.subAxisOf(ChannelId::x),
 	    !horizontal && stackInhibitingShape);
 
-	spacing.x = (horizontal && options.getChannels().anyAxisSet()
+	spacing.x = ((horizontal || lineOrCircle)
```

Comment:
This fixes the line/circle case, but this would fix the rectangle case as well:
```cpp
	spacing.x = (options.getChannels().anyAxisSet()
	                && channels.at(ChannelId::x).isDimension())
	              ? 1
	              : 0;
```

Reply:
I think it was a problem to even consider orientation here. Only the axis type (dimension or not) counts

Reply:
But this will fail on some test cases where it is intended to work like before. 
![kép](https://github.com/vizzuhq/vizzu-lib/assets/6457941/99c7c6ec-0378-411f-8750-916f925b59fa)
![kép](https://github.com/vizzuhq/vizzu-lib/assets/6457941/e54c2ab8-a2d6-441f-98ac-da2af8ed3982)
![kép](https://github.com/vizzuhq/vizzu-lib/assets/6457941/abf1a686-0981-4738-b159-427399a6186c)

Reply:
Yep, they should look like as they look on the right side. ->

Reply:
Oh, I see. The result is better, but the transition is problematic!